src/StreamJsonRpc/JsonRpcEventSource.cs

```diff
@@ -0,0 +1,312 @@
+
+        /// <summary>
+        /// The event ID for the <see cref="GenerateProxyUsingStreamsAndStartListeningStart(string)"/>.
+        /// </summary>
+        private const int GenerateProxyUsingStreamsAndStartListeningStartEvent = 3;
+
+        /// <summary>
+        /// The event ID for the <see cref="GenerateProxyAndStartListeningStop(string)"/>.
+        /// </summary>
+        private const int GenerateProxyUsingStreamsAndStartListeningStopEvent = 4;
+
+        /// <summary>
+        /// The event ID for the <see cref="GenerateProxyAndStartListeningStart(string)"/>.
+        /// </summary>
+        private const int GenerateProxyAndStartListeningStartEvent = 5;
+
+        /// <summary>
+        /// The event ID for the <see cref="GenerateProxyAndStartListeningStop(string)"/>.
+        /// </summary>
+        private const int GenerateProxyAndStartListeningStopEvent = 6;
+
+        /// <summary>
+        /// The event ID for the <see cref="GenerateProxyStart(string)"/>.
+        /// </summary>
+        private const int GenerateProxyStartEvent = 7;
+
+        /// <summary>
+        /// The event ID for the <see cref="GenerateProxyStop(string)"/>.
+        /// </summary>
+        private const int GenerateProxyStopEvent = 8;
+
+        /// <summary>
+        /// The event ID for the <see cref="MessageReceived(string)"/>.
+        /// </summary>
+        private const int MessageReceivedEvent = 20;
+
+        /// <summary>
+        /// The event ID for the <see cref="MessageSent(string)"/>.
+        /// </summary>
+        private const int MessageSentEvent = 21;
+
+        /// <summary>
+        /// The event ID for the <see cref="LogError(string)"/>.
+        /// </summary>
+        private const int LogErrorEvent = 22;
+
+        /// <summary>
+        /// The event ID for the <see cref="LogCriticalError(string)"/>.
+        /// </summary>
+        private const int LogCriticalErrorEvent = 23;
+
+        /// <summary>
+        /// The event ID for the <see cref="LogInformation(string)"/>.
+        /// </summary>
+        private const int LogInformationalEvent = 24;
+
+        /// <summary>
+        /// The event ID for the <see cref="LogWarning(string)"/>.
+        /// </summary>
+        private const int LogWarningEvent = 25;
+
+        /// <summary>
+        /// The event ID for the <see cref="ConnectionClosing(string)"/>.
+        /// </summary>
+        private const int ConnectionClosingEvent = 26;
+
+        /// <summary>
+        /// The event ID for the <see cref="LocalMethodAdded(string)"/>.
+        /// </summary>
+        private const int LocalMethodAddedEvent = 27;
+
+        /// <summary>
+        /// Creates a string representation of arguments for logging.
+        /// </summary>
+        /// <param name="arguments">A single argument or an array of arguments.</param>
+        /// <returns>String representation of arguments.</returns>
+        public static string GetArgumentsString(object arguments)
```

Comment:
nit: avoid `public` for what are really internal members (since the containing type is internal).

Reply:
Done